compiler/src/dmd/backend/dt.d

```diff
@@ -309,29 +309,29 @@ nothrow:
      *  _align = log2() of byte alignment of pointed-to data
      */
     @trusted
-    void abytes(tym_t ty, uint offset, uint size, const(char)* ptr, uint nzeros, ubyte _align)
+    void abytes(tym_t ty, uint offset, const(char)[] data, uint nzeros, ubyte _align)
     {
         dt_t *dt = dt_calloc(DT.abytes);
-        const n = size + nzeros;
-        assert(n >= size);      // overflow check
+        const n = data.length + nzeros;
+        assert(n >= data.length);      // overflow check
         dt.DTnbytes = n;
         dt.DTpbytes = cast(byte *) mem_malloc(n);
         dt.Dty = cast(ubyte)ty;
         dt.DTalign = _align;
         dt.DTabytes = offset;
-        memcpy(dt.DTpbytes,ptr,size);
+        memcpy(dt.DTpbytes, data.ptr, data.length);
         if (nzeros)
-            memset(dt.DTpbytes + size, 0, nzeros);
+            memset(dt.DTpbytes + data.length, 0, nzeros);
```

Comment:
can we use array assign and make this not `@trusted` now?

Reply:
I added array assign, but the dt.DTpbytes is still a malloced pointer that's written to so it's still not `@safe`

Reply:
OK